channeldb/graph.go

```diff
@@ -2118,13 +2120,37 @@ func (c *ChannelGraph) FilterKnownChanIDs(chanIDs []uint64) ([]uint64, error) {
 
 			// If the edge is a known zombie, skip it.
 			if zombieIndex != nil {
-				isZombie, _, _ := isZombieEdge(zombieIndex, cid)
-				if isZombie {
+				isZombie, _, _ := isZombieEdge(
+					zombieIndex, scid,
+				)
+
+				isStillZombie := isZombieChan(
+					info.Node1UpdateTimestamp,
+					info.Node2UpdateTimestamp,
+				)
+
+				switch {
+				// If the edge is a known zombie and if we
+				// would still consider it a zombie given the
+				// latest update timestamps, then we skip this
+				// channel.
+				case isZombie && isStillZombie:
 					continue
+
+				// Otherwise, if we have marked it as a zombie
+				// but the latest update timestamps could bring
+				// it back from the dead, then we mark it alive,
+				// and we let it be added to the set of IDs to
+				// query our peer for.
+				case isZombie && !isStillZombie:
+					err := c.markEdgeLive(tx, scid)
```

Comment:
Isn't marking the edge live here a little premature?  We haven't requested and checked the new channel update yet.

Once we do process the channel update, the existing code should already mark the edge live.

Reply:
Trying to think about the case where a random peer can make us remove arbitrary entries from our zombie index bucket w/o having more recent updates for these channels. It doesn't seem like an issue as if we ever received the updates after removing them from the zombie index, we'd just go through the validation pipeline again, but figured I'd point it out

Reply:
If truly dead edges are marked live, could that degrade payment success rates?

Reply:
This code from `MarkEdgeLive` concerns me:

https://github.com/lightningnetwork/lnd/blob/f005b248ced9bcd2316c0cbe7c9c25298796513c/channeldb/graph.go#L3529-L3542

It looks like we're activating the edge for pathfinding.

Reply:
`c.FetchChanInfos` here will return an empty list though since we would have deleted the edge when we pruned this channel via [`delChannelEdge`](https://github.com/lightningnetwork/lnd/blob/f005b248ced9bcd2316c0cbe7c9c25298796513c/channeldb/graph.go#L2379)

 which first [deletes the edge](https://github.com/lightningnetwork/lnd/blob/f005b248ced9bcd2316c0cbe7c9c25298796513c/channeldb/graph.go#L2435) 

before [marking it as a zombie](https://github.com/lightningnetwork/lnd/blob/master/channeldb/graph.go#L2458)

Reply:
so `MarkEdgeLive` will only load the channel edge info back into our graph cache if we still have the channel edge on disk which we wont in this case

Reply:
Makes sense, thanks for explaining!